Mathlib/Data/Seq/Seq.lean

```diff
@@ -262,16 +269,18 @@ theorem tail_nil : tail (nil : Seq α) = nil :=
 
 @[simp]
 theorem tail_cons (a : α) (s) : tail (cons a s) = s := by
-  cases' s with f al <;> apply Subtype.eq <;> dsimp [tail, cons] <;> rw [Stream'.tail_cons]
+  cases' s with f al ; apply Subtype.eq ; dsimp [tail, cons] ; rw [Stream'.tail_cons]
 #align stream.seq.tail_cons Stream'.Seq.tail_cons
 
 @[simp]
-theorem nth_tail (s : Seq α) (n) : nth (tail s) n = nth s (n + 1) :=
+theorem get?_tail (s : Seq α) (n) : get? (tail s) n = get? s (n + 1) :=
   rfl
-#align stream.seq.nth_tail Stream'.Seq.nth_tail
+#align stream.seq.nth_tail Stream'.Seq.get?_tail
 
-/-- Recursion principle for sequences, compare with `list.rec_on`. -/
-def recOn {C : Seq α → Sort v} (s : Seq α) (h1 : C nil) (h2 : ∀ x s, C (cons x s)) : C s := by
+/-- Recursion principle for sequences, compare with `List.recOn`. -/
+-- porting note: TODO: use match rather than `induction'`
```

Comment:
You should be able to use the `rec_eq_recC` trick we use elsewhere to make this computable.

Reply:
I'm confused how this will work here since we aren't recursing on `Seq` itself, but on `destruct s`...